src/mtl/include/mtl/MTLFormula.hpp

```diff
@@ -63,12 +63,50 @@ operator<<(std::ostream &out, const logic::MTLFormula<APType> &f)
 	case LOP::TRUE: out << u8"⊤"; break;
 	case LOP::FALSE: out << u8"⊥"; break;
 	case LOP::AP: out << f.get_atomicProposition(); break;
-	case LOP::LAND:
-		out << "(" << f.get_operands().front() << " ∧ " << f.get_operands().back() << ")";
+	case LOP::LAND: {
+		if (f.get_operands().empty()) {
+			out << u8"⊤";
+			break;
+		}
+		if (f.get_operands().size() == 1) {
+			out << f.get_operands().front();
+			break;
+		}
+		out << "(";
+		bool first = true;
+		for (const auto &op : f.get_operands()) {
+			if (first) {
+				first = false;
+			} else {
+				out << " ∧ ";
+			}
+			out << op;
+		}
```

Comment:
out of curiosity, is there not some fmt-function "join" to do this?

Reply:
Yes, but it requires custom a `fmt::formatter` for the underlying type. That would be an alternative, but it didn't seem necessary here